marshaling_test.go

```diff
@@ -266,17 +266,56 @@ func (s *Zuite) TestStructScan_slicesNotSupported() {
 	require.EqualError(s.T(), err, "struct field Texts: cannot StructScan slices (yet)")
 }
 
-type allTypesStruct struct {
-	Ws   *allTypesStruct `ws:"ws"`
-	Num0 int             `ws:"num_0"`
+func (s *Zuite) TestStructScan_refsPtr() {
```

Comment:
You'd also need to support cases such as
```
type worksheet self {
  1:me self
  2:my_friends []self
}
```
with graphs of the sort
```
ws.self = ws
ws.my_friends = [friend1, friend2]
friend1.my_friends = [ws, friend2]
friend2.my_friends = [ws, friend1]
```
The pattern followed throughout worksheets is as follows
- https://github.com/helloeave/worksheets/blob/master/marshaling.go#L52
- https://github.com/helloeave/worksheets/blob/master/dbstore.go#L403
- https://github.com/helloeave/worksheets/blob/master/clone.go#L32

Reply:
Yep, I added a comment above to do cycle checking as a follow up. Happy to add it here if you feel they should go together. Thanks for the references!